packages/next/src/server/base-server.ts

```diff
@@ -519,7 +519,23 @@ export default abstract class Server<ServerOptions extends Options = Options> {
   ): Promise<void> {
     return getTracer().trace(
       BaseServerSpan.handleRequest,
-      async () => await this.handleRequestImpl(req, res, parsedUrl)
+      {
+        tracerName: [req.method, req.url].join(' '),
+        kind: SpanKind.SERVER,
+        attributes: {
+          'http.method': req.method,
+          'http.target': req.url,
+        },
+      },
+      async (span) => {
+        try {
+          await this.handleRequestImpl(req, res, parsedUrl)
```

Comment:
maybe we should move the tracing to be inside handleRequestImpl instead

Reply:
Isn't the only reason for splitting `handleRequest` and `handleRequestImpl` just tracing?

I don't mind inlining `handleRequestImpl` it if you think it's better.

Reply:
sorry I forgot I had done this! What you can do is annotate the span from inside the function via getActiveSpan

Reply:
:+1: I know that.

But a) this could cause issues if you enable verbose logging since that would change the annotated span (not here though).

B) `try/catch` here seems better than trying to patch `activeSpan` whenever we set the status code.

Reply:
This was also suggested in another PR, but that wouldn't work.

> fix types of trace callback (mark it as optional)
I tried doing it properly with a mocked span, but it's better to be explicit because using mock span as parentSpan would silence the whole tree.

Reply:
It will work as long as we don't pass it explicitly as `parentSpan` somewhere. We could check for that and ignore the parent span Prop if it's noop span, but that feels like like too much magic for very little gain.